### ISSUE TITLE
Support the "attrs" library

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#1698]
Many projects (Twisted, Hypothesis, pytest, my own, and probably more) started using [attrs](http://www.attrs.org/) because it makes it much more pleasant to write data classes.

However, pylint doesn't understand attrs' handling of default values:

```python
import attr

@attr.s
class Foo:

    d = attr.ib(attr.Factory(dict))

f = Foo()
f.d['answer'] = 42
```

outputs:

```
E:  9, 0: 'f.d' does not support item assignment (unsupported-assignment-operation)
```

Unfortunately, I can't find a way to turn this off selectively either, e.g. by adding `Foo` to `ignored-classes`.

**Comment by PCManticore** [comment]
This probably needs a [brain plugin](https://github.com/PyCQA/astroid/tree/c2e8ca524793905018e0e4adc5e29e1d3b0446c6/astroid/brain).
Also unfortunately `ignored-classes` can only be applied for a handful of errors, I think `unsupported-assignment-operation` doesn't fall in that category (`ignored-classes` is mostly for `no-member` and a bunch of others).

**Comment by tedbrandston** [comment]
So is this a thing where we should just be waiting for the next release of pylint? Or is there something we can do to get this working right now?

Thanks!

**Comment by The-Compiler** [comment]
It's part of astroid 1.6.0 which already has been released.

**Comment by dfee** [comment]
Why is this closed? It's not supported, at least not on:
• astroid `2.0.0.dev4`,
• pylint `2.0.0`,
• attrs `18.1.0`

**Comment by zadamah** [comment]
I think that that `attr` decorators with no arguments are supported
My pull request #479 for the case with arguments was merged to master but until recently it did not make its way to a release. @The-Compiler can you help us on this ?

**Comment by PCManticore** [comment]
Folks, please open separate issues. Some `attrs` support landed in pylint but we cannot support everything out of the box unless we have separate issues for each particular thing that is not working. Also do take in account that we're all volunteers and we might not use `attrs` on a daily basis so there's a chance that we might not know what needs to be supported. Another thing is that asking why is this closed is usually not helpful, as most of the time we close issues either because there is no plan to fix them or (some partial) support for that particular feature landed already in the project.

**Comment by The-Compiler** [comment]
@zadamah I don't understand what you want me to help with? It looks like with https://github.com/PyCQA/astroid/pull/479 this should work, and @dfee seems to be using an older astroid release.

@dfee Like @PCManticore said, if you're still having issues with an up-to-date pylint/astroid, please open a separate issue, with some more information than "it's not supported" - no issues here with attrs ever since this was closed.

**Comment by zadamah** [comment]
Actually I replied to @dfee here but it is about PYCQA/asstroid#479:
* it was merged to astroid master on January
* but it was not merged to the release branch 2.0 (last I checked)
* so what I am not sure of is what can we do so that this feature is on the next astroid release and then in pylint as well ?

And obviously I completely agree what @PCManticore is saying. pylint is actually one of my favorite Python tools. I just want this feature to be available to everyone.

**Comment by zadamah** [comment]
Yes I was referring to branch "2.0" from astroid, it got me a little confused.
And you are right, I haven't checked for while, but now I can see the changes in *brain_attrs.py* after upgrading to pylint 2.1.0 (and astroid 2.0.2).
Thank you @The-Compiler & @PCManticore.
And @dfee, I think that your issue could be solved by upgrading the latest pylint release.
